lib/collector.js

```diff
@@ -28,10 +28,10 @@ class Collector {
     if (typeof timeout === 'number' && timeout > 0) {
       this.#timeout = AbortSignal.timeout(timeout);
       this.#signal = AbortSignal.any([this.#signal, this.#timeout]);
-      this.#timeout.addEventListener('abort', () => {
+      this.#signal.addEventListener('abort', () => {
         if (Object.keys(this.defaults).length > 0) this.#default();
         if (this.done) return;
```

Comment:
```suggestion
        const reason = this.#signal.reason
        if (reason.endsWith('timeout') && Object.keys(this.defaults).length > 0) this.#default();
        if (this.done) return;
```
After this fix, default values will be used after all abort signals. So we should check if this is a signal from timer.

Reply:
I'm a little skeptical about using ```.endsWith``` in this case. If a comparison is needed there, I think it makes sense to get the error name via ```dc.signal.reason.name``` and compare like ```dc.signal.reason.name === 'TimeoutError'```, although I don't think this is necessary to check.